src/operations/execute_operation.ts

```diff
@@ -150,6 +151,16 @@ function executeWithServerSelection(
     session.unpin();
   }
 
+  let selector: ReadPreference | ServerSelector;
+
+  // If operation should try to write to secondary use the custom server selector
+  // otherwise provide the read preference.
+  if (operation.trySecondaryWrite) {
```

Comment:
In the case of a potential write to a secondary, we use our new server selector instead of passing the read preference directly to `selectServer`.